Extensions/TileMap/JsExtension.js

```diff
@@ -591,6 +591,383 @@ const defineTileMap = function (extension, _, gd) {
+    .addParameter('yesorno', _('Flip horizontally'), '', false)
+    .setDefaultValue('false')
+    .addParameter('yesorno', _('Flip vertically'), '', false)
+    .setDefaultValue('false')
+    .setFunctionName('flipTileAtSceneCoordinates');
+
+  object
+    .addAction(
+      'RemoveTileAtSceneCoordinates',
+      _('Remove tile (scene)'),
+      _('Remove the tile at the scene coordinates'),
+      _('Remove tile at scene coordinates _PARAM1_;_PARAM2_'),
+      '',
+      'JsPlatform/Extensions/tile_map.svg',
+      'JsPlatform/Extensions/tile_map.svg'
+    )
+    .addParameter('object', _('Tile map'), 'SimpleTileMap', false)
+    .addParameter('number', _('Position X'), '', false)
+    .addParameter('number', _('Position Y'), '', false)
+    .getCodeExtraInformation()
+    .setFunctionName('removeTileAtSceneCoordinates');
+
+  object
+    .addExpressionAndConditionAndAction(
+      'number',
+      'TileIdAtGrid',
+      _('Tile identifier (grid)'),
+      _('the id of the tile at the grid coordinates'),
+      _('the tile id at grid coordinates _PARAM2_;_PARAM3_'),
+      '',
+      'JsPlatform/Extensions/tile_map.svg'
+    )
+    .addParameter('object', _('Tile map'), 'SimpleTileMap', false)
+    .useStandardParameters('number', gd.ParameterOptions.makeNewOptions())
+    .addParameter('number', _('Grid X'), '', false)
+    .addParameter('number', _('Grid Y'), '', false)
+    .setFunctionName('setTileAtGridCoordinates')
+    .setGetter('getTileAtGridCoordinates');
+
+  object
+    .addAction(
+      'FlipTileAtGridCoordinates',
+      _('Flip tile (grid)'),
+      _('Flip tile at grid coordinates.'),
+      _(
+        'Flip tile at grid coordinates _PARAM1_;_PARAM2_: flip horizontally _PARAM3_, flip vertically _PARAM4_'
+      ),
+      '',
+      'JsPlatform/Extensions/tile_map.svg',
+      'JsPlatform/Extensions/tile_map.svg'
+    )
+    .addParameter('object', _('Tile map'), 'SimpleTileMap', false)
+    .addParameter('number', _('Grid X'), '', false)
+    .addParameter('number', _('Grid Y'), '', false)
+    .addParameter('yesorno', _('Flip horizontally'), '', false)
+    .setDefaultValue('false')
+    .addParameter('yesorno', _('Flip vertically'), '', false)
+    .setDefaultValue('false')
+    .setFunctionName('flipTileAtGridCoordinates');
+
+  object
+    .addAction(
+      'RemoveTileAtGridCoordinates',
+      _('Remove tile (grid)'),
+      _('Remove the tile at the grid coordinates'),
+      _('Remove tile at grid coordinates _PARAM1_;_PARAM2_'),
+      '',
+      'JsPlatform/Extensions/tile_map.svg',
+      'JsPlatform/Extensions/tile_map.svg'
+    )
+    .addParameter('object', _('Tile map'), 'SimpleTileMap', false)
+    .addParameter('number', _('Grid X'), '', false)
+    .addParameter('number', _('Grid Y'), '', false)
+    .getCodeExtraInformation()
+    .setFunctionName('removeTileAtGridCoordinates');
+
+  object
+    .addCondition(
+      'IsTileFlippedHorizontallyAtSceneCoordinates',
```

Comment:
To me `Horizontal` and `Vertical` for flipping is very hard to understand because I picture the mirror which is vertical for horizontal flipping so I get confused.
```suggestion
      'IsTileFlippedOnXAtPosition',
```